File: src/main/java/com/uapp_llc/service/ColumnService.java
package com.uapp_llc.service;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

import com.uapp_llc.model.Column;

public interface ColumnService {

  Column create(String name);

  Column update(Long id, String name);

  Column changeIndex(Long id, Integer index);

  Column find(Long id);

  Page<Column> findAll(Pageable pageable);

  void delete(Long id);

}

File: src/main/kotlin/com/uapp_llc/service/ColumnService.kt
package com.uapp_llc.service

import org.springframework.data.domain.Page
import org.springframework.data.domain.Pageable
import com.uapp_llc.model.Column

interface ColumnService {

  fun create(name: String): Column

  fun update(id: Long, name: String?): Column

  fun changeIndex(id: Long, index: Int): Column

  fun find(id: Long): Column

  fun findAll(pageable: Pageable): Page<Column>

  fun delete(id: Long)

}
